Cython/Compiler/ParseTreeTransforms.py

```diff
@@ -2239,6 +2239,25 @@ def visit_IndexNode(self, node):
             node = node.base
         return node
 
+class ReplacePropertyNode(CythonTransform):
+    def visit_CFuncDefNode(self, node):
+        if not node.decorators:
+            return node
+        # transform @property decorators on ctypedef class functions
+        for decorator_node in node.decorators[::-1]:
+            decorator = decorator_node.decorator
+            if decorator.is_name and decorator.name == 'property':
+                if len(node.decorators) > 1:
+                    return self._reject_decorated_property(node, decorator_node)
+                # Mark the node as a cgetter
+                node.type.is_cgetter = True
+                # Add a func_cname to be output instead of the attribute
+                node.entry.func_cname = node.body.stats[0].value.function.name
+                # done - remove the decorator node
+                node.decorators.remove(decorator_node)
+                return [node]
+        return [node]
```

Comment:
I think returning the node (and not a single element list) should be fine here. Actually, I'd a) use `break` instead of duplicating the `return` code, and b) use a search loop that exits and then does stuff, rather than doing everything in the loop. (And c) use a helper to find the right decorator.)